packages/core/src/App/Components/Elements/Errors/error-boundary.jsx

```diff
@@ -1,32 +1,36 @@
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import React from 'react';
-import { connect } from 'Stores/connect';
+import { observer, useStore } from '@deriv/stores';
 import ErrorComponent from './index';
 
-class ErrorBoundary extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = { hasError: false };
-    }
+const ErrorBoundary = observer(({ children }) => {
+    const root_store = useStore();
+    const [hasError, setHasError] = useState(false);
+    const [error, setError] = useState(null);
+    const [info, setInfo] = useState(null);
 
-    componentDidCatch = (error, info) => {
-        if (window.TrackJS) window.TrackJS.console.log(this.props.root_store);
+    const handleError = (errorMsg, infoMsg) => {
+        if (window.TrackJS) window.TrackJS.console.log(root_store);
 
-        this.setState({
-            hasError: true,
-            error,
-            info,
-        });
+        setHasError(true);
+        setError(errorMsg);
+        setInfo(infoMsg);
     };
+    useEffect(() => {
+        window.addEventListener('error', () => {
+            handleError(error, info);
+        });
+        return () =>
+            window.removeEventListener('error', () => {
+                handleError(error, info);
+            });
```

Comment:
```suggestion
        const errorHandler = () => {
            handleError(error, info);
        }
        window.addEventListener('error', errorHandler);
        return () =>
            window.removeEventListener('error', errorHandler);
```

@yaswanth-deriv, why could you be passing a different functions for the `addEventListener` and `removeEventListener`? 🤔